### PR TITLE
Transform Widget alignment is AlignmentGeometry

### PR DESCRIPTION
**Pull request by HansMuller**
Loosen the type of the Transform widget's alignment parameter to AlignmentGeometry.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -1632,16 +1634,35 @@ class RenderTransform extends RenderProxyBox {
   ///
   /// This is equivalent to setting an origin based on the size of the box.
   /// If it is specified at the same time as an offset, both are applied.
-  Alignment get alignment => _alignment;
-  Alignment _alignment;
-  set alignment(Alignment value) {
-    assert(value == null || (value.x != null && value.y != null));
+  ///
+  /// A [AlignmentDirectional.start] value is the same as an [Alignment]
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: An

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 2]
```diff
@@ -1632,16 +1634,35 @@ class RenderTransform extends RenderProxyBox {
...
+  ///
+  /// This may be changed to null, but only after [alignment] has been changed
+  /// to a value that does not depend on the direction.
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value)
+      return;
+    _textDirection = value;
+    markNeedsPaint();
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: you want to use the markNeedsResolve pattern, see some of the other classes here or in shifted_box

File: packages/flutter/lib/src/rendering/proxy_box.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: There were a lot of these, I think I've got them all.

File: packages/flutter/lib/src/rendering/proxy_box.dart (hunk 2 above)
Comment (HansMuller) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: Transform allows alignment to be null which complicated applying the pattern. Plus the cost of resolving an Alignment is small.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**